File: kafka/go/pkg/vehicle/vehicle-position.go
package vehicle

import "encoding/json"

type VehiclePositionRoot struct {
	VP VehiclePosition `json:"VP"`
}

type VehiclePosition struct {
	Veh int

	Desi string
	Dir  string
	Oper int
	lat  float64
	long float64
}

func (i VehiclePosition) MarshalBinary() (data []byte, err error) {
	bytes, err := json.Marshal(i)
	return bytes, err
}

File: kafka/java/src/main/java/com/vehicle/model/VehiclePosition.java
package com.vehicle.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.redis.core.RedisHash;

@RedisHash()
public class VehiclePosition {

  @Id
  private int veh;

  
  private String desi;
  private String dir;
  private String oper;

  
  public VehiclePosition() {
  }

  public String getDesi() {
    return desi;
  }

  public void setDesi(String desi) {
    this.desi = desi;
  }

  public String getDir() {
    return dir;
  }

  public void setDir(String dir) {
    this.dir = dir;
  }

  public String getOper() {
    return oper;
  }

  public void setOper(String oper) {
    this.oper = oper;
  }

  public int getVeh() {
    return veh;
  }

  public void setVeh(int veh) {
    this.veh = veh;
  }

}
